<?xml version="1.0" encoding="UTF-8"?>
<robot generator="Robot 7.1.1 (Python 3.13.1 on win32)" generated="2025-03-12T00:41:37.560382" rpa="false" schemaversion="5">
<suite id="s1" name="Test" source="D:\sprint1\git-group-repository-group-3-sec-2-v-2\Test\testRobotSprint3\test.robot">
<test id="s1-t1" name="Test SQL Injection Security" line="9">
<kw name="GET" owner="RequestsLibrary">
<msg time="2025-03-12T00:41:39.234229" level="INFO">GET Request : url=https://projectsoften.cpkkuhost.com/sql 
 path_url=/sql 
 headers={'User-Agent': 'python-requests/2.32.3', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} 
 body=None 
 </msg>
<msg time="2025-03-12T00:41:39.234437" level="INFO">GET Response : url=https://projectsoften.cpkkuhost.com/sql 
 status=406, reason=Not Acceptable 
 headers={'Date': 'Tue, 11 Mar 2025 17:41:37 GMT', 'Server': 'Apache', 'Content-Length': '226', 'Keep-Alive': 'timeout=5, max=75', 'Connection': 'Keep-Alive', 'Content-Type': 'text/html; charset=iso-8859-1'} 
 body=&lt;head&gt;&lt;title&gt;Not Acceptable!&lt;/title&gt;&lt;/head&gt;&lt;body&gt;&lt;h1&gt;Not Acceptable!&lt;/h1&gt;&lt;p&gt;An appropriate representation of the requested resource could not be found on this server. This error was generated by Mod_Security.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt; 
 </msg>
<msg time="2025-03-12T00:41:39.234663" level="FAIL">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/sql</msg>
<var>${response}</var>
<arg>${BASE_URL}/sql</arg>
<doc>Sends a GET request.</doc>
<status status="FAIL" start="2025-03-12T00:41:38.292671" elapsed="1.106292">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/sql</status>
</kw>
<kw name="Should Be Equal As Strings" owner="BuiltIn">
<arg>${response.status_code}</arg>
<arg>200</arg>
<doc>Fails if objects are unequal after converting them to strings.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:39.399865" elapsed="0.000090"/>
</kw>
<kw name="Should Contain" owner="BuiltIn">
<arg>${response.text}</arg>
<arg>SQL Security Test Passed</arg>
<doc>Fails if ``container`` does not contain ``item`` one or more times.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:39.400302" elapsed="0.000027"/>
</kw>
<status status="FAIL" start="2025-03-12T00:41:38.291299" elapsed="1.109451">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/sql</status>
</test>
<test id="s1-t2" name="Test XSS Security" line="14">
<kw name="GET" owner="RequestsLibrary">
<msg time="2025-03-12T00:41:40.267901" level="INFO">GET Request : url=https://projectsoften.cpkkuhost.com/xss 
 path_url=/xss 
 headers={'User-Agent': 'python-requests/2.32.3', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} 
 body=None 
 </msg>
<msg time="2025-03-12T00:41:40.268036" level="INFO">GET Response : url=https://projectsoften.cpkkuhost.com/xss 
 status=406, reason=Not Acceptable 
 headers={'Date': 'Tue, 11 Mar 2025 17:41:38 GMT', 'Server': 'Apache', 'Content-Length': '226', 'Keep-Alive': 'timeout=5, max=75', 'Connection': 'Keep-Alive', 'Content-Type': 'text/html; charset=iso-8859-1'} 
 body=&lt;head&gt;&lt;title&gt;Not Acceptable!&lt;/title&gt;&lt;/head&gt;&lt;body&gt;&lt;h1&gt;Not Acceptable!&lt;/h1&gt;&lt;p&gt;An appropriate representation of the requested resource could not be found on this server. This error was generated by Mod_Security.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt; 
 </msg>
<msg time="2025-03-12T00:41:40.278756" level="FAIL">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/xss</msg>
<var>${response}</var>
<arg>${BASE_URL}/xss</arg>
<doc>Sends a GET request.</doc>
<status status="FAIL" start="2025-03-12T00:41:39.402828" elapsed="0.877444">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/xss</status>
</kw>
<kw name="Should Be Equal As Strings" owner="BuiltIn">
<arg>${response.status_code}</arg>
<arg>200</arg>
<doc>Fails if objects are unequal after converting them to strings.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:40.280819" elapsed="0.000044"/>
</kw>
<kw name="Should Contain" owner="BuiltIn">
<arg>${response.text}</arg>
<arg>XSS Security Test Passed</arg>
<doc>Fails if ``container`` does not contain ``item`` one or more times.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:40.281169" elapsed="0.000028"/>
</kw>
<status status="FAIL" start="2025-03-12T00:41:39.402259" elapsed="0.879321">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/xss</status>
</test>
<test id="s1-t3" name="Test DDoS Security" line="19">
<kw name="GET" owner="RequestsLibrary">
<msg time="2025-03-12T00:41:41.089295" level="INFO">GET Request : url=https://projectsoften.cpkkuhost.com/ddos 
 path_url=/ddos 
 headers={'User-Agent': 'python-requests/2.32.3', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} 
 body=None 
 </msg>
<msg time="2025-03-12T00:41:41.089427" level="INFO">GET Response : url=https://projectsoften.cpkkuhost.com/ddos 
 status=406, reason=Not Acceptable 
 headers={'Date': 'Tue, 11 Mar 2025 17:41:39 GMT', 'Server': 'Apache', 'Content-Length': '226', 'Keep-Alive': 'timeout=5, max=75', 'Connection': 'Keep-Alive', 'Content-Type': 'text/html; charset=iso-8859-1'} 
 body=&lt;head&gt;&lt;title&gt;Not Acceptable!&lt;/title&gt;&lt;/head&gt;&lt;body&gt;&lt;h1&gt;Not Acceptable!&lt;/h1&gt;&lt;p&gt;An appropriate representation of the requested resource could not be found on this server. This error was generated by Mod_Security.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt; 
 </msg>
<msg time="2025-03-12T00:41:41.089606" level="FAIL">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/ddos</msg>
<var>${response}</var>
<arg>${BASE_URL}/ddos</arg>
<doc>Sends a GET request.</doc>
<status status="FAIL" start="2025-03-12T00:41:40.283710" elapsed="0.807143">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/ddos</status>
</kw>
<kw name="Should Be Equal As Strings" owner="BuiltIn">
<arg>${response.status_code}</arg>
<arg>200</arg>
<doc>Fails if objects are unequal after converting them to strings.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:41.091421" elapsed="0.000042"/>
</kw>
<kw name="Should Contain" owner="BuiltIn">
<arg>${response.text}</arg>
<arg>DDoS Security Test Passed</arg>
<doc>Fails if ``container`` does not contain ``item`` one or more times.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:41.091729" elapsed="0.000035"/>
</kw>
<status status="FAIL" start="2025-03-12T00:41:40.283232" elapsed="0.809292">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/ddos</status>
</test>
<test id="s1-t4" name="Test Brute Force Security" line="24">
<kw name="GET" owner="RequestsLibrary">
<msg time="2025-03-12T00:41:41.888867" level="INFO">GET Request : url=https://projectsoften.cpkkuhost.com/brute 
 path_url=/brute 
 headers={'User-Agent': 'python-requests/2.32.3', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} 
 body=None 
 </msg>
<msg time="2025-03-12T00:41:41.889000" level="INFO">GET Response : url=https://projectsoften.cpkkuhost.com/brute 
 status=406, reason=Not Acceptable 
 headers={'Date': 'Tue, 11 Mar 2025 17:41:40 GMT', 'Server': 'Apache', 'Content-Length': '226', 'Keep-Alive': 'timeout=5, max=75', 'Connection': 'Keep-Alive', 'Content-Type': 'text/html; charset=iso-8859-1'} 
 body=&lt;head&gt;&lt;title&gt;Not Acceptable!&lt;/title&gt;&lt;/head&gt;&lt;body&gt;&lt;h1&gt;Not Acceptable!&lt;/h1&gt;&lt;p&gt;An appropriate representation of the requested resource could not be found on this server. This error was generated by Mod_Security.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt; 
 </msg>
<msg time="2025-03-12T00:41:41.889265" level="FAIL">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/brute</msg>
<var>${response}</var>
<arg>${BASE_URL}/brute</arg>
<doc>Sends a GET request.</doc>
<status status="FAIL" start="2025-03-12T00:41:41.094171" elapsed="0.796298">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/brute</status>
</kw>
<kw name="Should Be Equal As Strings" owner="BuiltIn">
<arg>${response.status_code}</arg>
<arg>200</arg>
<doc>Fails if objects are unequal after converting them to strings.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:41.891189" elapsed="0.000036"/>
</kw>
<kw name="Should Contain" owner="BuiltIn">
<arg>${response.text}</arg>
<arg>Brute Force Test Passed</arg>
<doc>Fails if ``container`` does not contain ``item`` one or more times.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:41.891505" elapsed="0.000023"/>
</kw>
<status status="FAIL" start="2025-03-12T00:41:41.093733" elapsed="0.798146">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/brute</status>
</test>
<test id="s1-t5" name="Test Failed Login Security" line="29">
<kw name="GET" owner="RequestsLibrary">
<msg time="2025-03-12T00:41:42.682365" level="INFO">GET Request : url=https://projectsoften.cpkkuhost.com/failed-login 
 path_url=/failed-login 
 headers={'User-Agent': 'python-requests/2.32.3', 'Accept-Encoding': 'gzip, deflate', 'Accept': '*/*', 'Connection': 'keep-alive'} 
 body=None 
 </msg>
<msg time="2025-03-12T00:41:42.682534" level="INFO">GET Response : url=https://projectsoften.cpkkuhost.com/failed-login 
 status=406, reason=Not Acceptable 
 headers={'Date': 'Tue, 11 Mar 2025 17:41:41 GMT', 'Server': 'Apache', 'Content-Length': '226', 'Keep-Alive': 'timeout=5, max=75', 'Connection': 'Keep-Alive', 'Content-Type': 'text/html; charset=iso-8859-1'} 
 body=&lt;head&gt;&lt;title&gt;Not Acceptable!&lt;/title&gt;&lt;/head&gt;&lt;body&gt;&lt;h1&gt;Not Acceptable!&lt;/h1&gt;&lt;p&gt;An appropriate representation of the requested resource could not be found on this server. This error was generated by Mod_Security.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt; 
 </msg>
<msg time="2025-03-12T00:41:42.682750" level="FAIL">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/failed-login</msg>
<var>${response}</var>
<arg>${BASE_URL}/failed-login</arg>
<doc>Sends a GET request.</doc>
<status status="FAIL" start="2025-03-12T00:41:41.893672" elapsed="0.790496">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/failed-login</status>
</kw>
<kw name="Should Be Equal As Strings" owner="BuiltIn">
<arg>${response.status_code}</arg>
<arg>200</arg>
<doc>Fails if objects are unequal after converting them to strings.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:42.684701" elapsed="0.000056"/>
</kw>
<kw name="Should Contain" owner="BuiltIn">
<arg>${response.text}</arg>
<arg>Failed Login Security Test Passed</arg>
<doc>Fails if ``container`` does not contain ``item`` one or more times.</doc>
<status status="NOT RUN" start="2025-03-12T00:41:42.685031" elapsed="0.000028"/>
</kw>
<status status="FAIL" start="2025-03-12T00:41:41.893241" elapsed="0.792229">HTTPError: 406 Client Error: Not Acceptable for url: https://projectsoften.cpkkuhost.com/failed-login</status>
</test>
<status status="FAIL" start="2025-03-12T00:41:37.563737" elapsed="5.123514"/>
</suite>
<statistics>
<total>
<stat pass="0" fail="5" skip="0">All Tests</stat>
</total>
<tag>
</tag>
<suite>
<stat pass="0" fail="5" skip="0" id="s1" name="Test">Test</stat>
</suite>
</statistics>
<errors>
</errors>
</robot>
